crates/core/src/lib.rs

```diff
@@ -32,10 +32,25 @@ pub use alloy_json_abi as json_abi;
 #[cfg(feature = "sol-types")]
 #[doc(inline)]
 pub use alloy_sol_types as sol_types;
-#[cfg(feature = "sol-types")]
+#[cfg(all(doc, feature = "sol-types"))]
 #[doc(no_inline)]
-pub use alloy_sol_types::sol;
+pub use sol_types::sol;
 
 #[cfg(feature = "rlp")]
 #[doc(inline)]
 pub use alloy_rlp as rlp;
+
+/// [`sol!`](alloy_sol_types::sol!) macro wrapper to route imports to the correct crate.
+///
+/// See [`sol!`](alloy_sol_types::sol!) for the actual macro documentation.
+#[cfg(all(not(doc), feature = "sol-types"))]
```

Comment:
`not(doc)` so that generated documentation (e.g. docs.rs) displays the `sol!` re-export instead of this wrapper.